1_fetch.R

```diff
@@ -0,0 +1,133 @@
+  # filenames mentioned in p1_ltmp_nml_list.rds
+  tar_target(
+    p1_meteo_filenames,
+    readRDS(p1_ltmp_nml_list.rds) %>%
+      purrr::map_chr('meteo_fl') %>%
+      unique() %>%
+      file.path('1_fetch/out', .)
+  ),
+  tar_target(
+    p1_meteo_files,
+    sb_download_if_needed(
+      sb_id = data_release_sb_id,
+      names = basename(p1_meteo_filenames),
+      destinations = p1_meteo_filenames,
+      outdated = sb_outdated, # can set outdated = ltmp_filenames to make this a target that only rebuilds on force
+      status_file = sb_status_csv_name), # rely on a non-target string so that updates to sb_status_csv don't trigger rebuilds here
+    packages = c('sbtools'),
+    format = 'file',
+    pattern = map(p1_meteo_filenames)
+  ),
+
+
+  # Download inflow-outflow data (just observations and SNTemp predictions,
+  # currently) from ScienceBase, using static branching over several files
+  tar_map(
+    values = tibble(
+      io_filenames = file.path('1_fetch/out', c('res_io_obs.feather', 'res_io_sntemp.feather')),
+      target_names = basename(io_filenames)),
+    names = starts_with('target_names'),
+    tar_target(
+      p1_io,
+      sb_download_if_needed(
+        sb_id = data_release_sb_id,
+        names = basename(io_filenames),
+        destinations = io_filenames,
+        outdated = sb_outdated, # can set outdated = io_filenames to make this a target that only rebuilds on force
+        status_file = sb_status_csv_name), # rely on a non-target string so that updates to sb_status_csv don't trigger rebuilds here
+      packages = c('sbtools'),
+      format = 'file')
+  ),
+
+  # Specify the source of inflow-outflow data (SNTemp predictions, currently)
+  # TODO - use res-temperature-data-prep to transfer this file via ScienceBase
+  # tar_target(
+  #   p1_inout_feather,
+  #   '../delaware-model-prep/9_collaborator_data/res/res_io_sntemp.feather',
+  #   format = 'file'),
+
+  # Get reservoir releases from SB at https://www.sciencebase.gov/catalog/item/5f6a287382ce38aaa2449131
+  # Because this is a different SB item from the res-temperature-data-sharing item, we can't pull only if needed very conveniently.
+  # TODO: pull reservoir release data from Sam's repo, add to res-temperature-data-sharing pipeline, then pull from there to here
+  tar_target(
+    p1_releases_csv,
+    {
+      dest <- '1_fetch/out/reservoir_releases.csv'
+      sb_secret_login()
+      sbtools::item_file_download(
+        sb_id = '5f6a287382ce38aaa2449131',
+        names = basename(dest),
+        destinations = dest,
+        overwrite_file = TRUE)
+      return(dest)
+    },
+    format = 'file'
+  ),
+
+  # Hand-code + calculate a set of edits to make to the nmls for each site.
+  # These edits may ultimately fit better within the lake-temperature-model-prep
+  # or delaware-model-prep pipelines
+  tar_target(
+    p1_nml_edits,
+    build_nml_edits(
+      p1_io_res_io_sntemp.feather,
```

Comment:
I'm confused about where `p1_io_res_io_sntemp.feather` comes from -- the commented out code about would generate a target named `p1_inout_feather`, but that target is commented out and seems to be referencing a feather file directly. It looks like in lines 81-96 the SNTemp predictions are downloaded to `1_fetch/out/res_io_sntemp.feather` -- is that the file that should be passed to `built_nml_edits` here?

Or wait - now I'm thinking this just has to do w/ `targets` syntax, and the target associated with the `1_fetch/out/res_io_sntemp.feather` file generated by the `tar_target` call nested in the `tar_map` call is `p1_io` concatenated with the target name `res_io_sntemp.feather`, which makes `p1_io_res_io_sntemp.feather`. Is that correct?

Reply:
You got it. I don't love this particular targets syntax, either - you don't get much control over how those targets are named, and I'm not aware of an easy way to get targets to list them for you, either.